gitclone/steperror.go

```diff
@@ -1,11 +1,178 @@
 package gitclone
 
-import "github.com/bitrise-io/bitrise-init/step"
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/bitrise-io/bitrise-init/step"
+)
+
+// DetailBuilder ...
+type DetailBuilder = func(...string) Detail
+
+// PatternErrorMatcher ...
+type PatternErrorMatcher struct {
+	defaultHandler DetailBuilder
+	handlers       map[string]DetailBuilder
+}
+
+func newPatternErrorMatcher(defaultHandler DetailBuilder, handlers map[string]DetailBuilder) *PatternErrorMatcher {
+	m := PatternErrorMatcher{
+		handlers:       handlers,
+		defaultHandler: defaultHandler,
+	}
+
+	return &m
+}
+
+// Run ...
+func (m *PatternErrorMatcher) Run(msg string) step.Recommendation {
+	for pattern, handler := range m.handlers {
+		re := regexp.MustCompile(pattern)
+		if re.MatchString(msg) {
+			matches := re.FindStringSubmatch((msg))
+
+			if len(matches) > 1 {
+				matches = matches[1:]
+			}
+
+			if matches != nil {
```

Comment:
i think if `re.MatchString(msg)` is true, matches can not be nil

Reply:
Probably you are right.